src/middlewared/middlewared/plugins/smb.py

```diff
@@ -1065,14 +1065,26 @@ async def query(self, filters=None, options=None):
             result = await self.middleware.call(
                 'sharing.smb.registry_query', filters, options
             )
-        else:
-            options['extend'] = self._config.datastore_extend
-            options['prefix'] = self._config.datastore_prefix
+            return result
+
+        if not filters:
+            filters = []
 
+        options = await self.get_options(options)
+        if options['extend']:
+            datastore_options = options.copy()
+            datastore_options.pop('count', None)
+            datastore_options.pop('get', None)
             result = await self.middleware.call(
-                'datastore.query', self._config.datastore, filters, options
+                'datastore.query', self._config.datastore, [], datastore_options
+            )
+            return await self.middleware.run_in_thread(
+                filter_list, result, filters, options
+            )
+        else:
+            return await self.middleware.call(
+                'datastore.query', self._config.datastore, filters, options,
             )
```

Comment:
Isnt it better to use super() here? Dont recall if this has already been fixed by anotehr PR

Reply:
Yeah, it was fixed in another PR by Waqar (happened right around thanksgiving).